### PR TITLE
StaticLLMPipeline: Handle CACHE_DIR for NPUW

### PR DESCRIPTION
**Pull request by smirnov-alexey**
E-149053

### CODE REVIEW

File: src/cpp/src/llm_pipeline_static.cpp [hunk 1]
```diff
@@ -571,8 +571,10 @@ ov::Tensor make_tensor_slice(ov::Tensor tensor, size_t dim, size_t start_pos, si
     return ov::Tensor(tensor, start_shape, end_shape);
 }
 
-void drop_cache_dir(ov::AnyMap& config) {
-    if (config.count("NPU_USE_NPUW") != 0u) {
+void set_npuw_cache_dir(ov::AnyMap& config) {
+    std::optional<std::string> cache_dir = get_option<std::string>(config, "CACHE_DIR");
+    if (config.count("NPU_USE_NPUW") != 0u && cache_dir) {
```
Comment (dmatveev) [inline src/cpp/src/llm_pipeline_static.cpp:576]: if NPUW is not used, but "CACHE_DIR" is specified, it should be passed as "CACHE_DIR" to the model.

**Review by dmatveev — COMMENTED** [review]
Need to think how this behavior changes once we introduce the blob compilation (weightless or full) and what'd happen in this case.

File: src/cpp/src/llm_pipeline_static.cpp (hunk 1 above)
Comment (smirnov-alexey) [reply src/cpp/src/llm_pipeline_static.cpp:576]: In this case `CACHE_DIR` is already in the config and this `if` won't trigger, what's the problem?

Comment (dmatveev) [reply src/cpp/src/llm_pipeline_static.cpp:576]: ah, ok - makes sense, thanks!

**Review by dmatveev — APPROVED** [review]

**Pull request merged**